code/game/objects/items/devices/traitordevices.dm

```diff
@@ -255,6 +255,16 @@ effective or pretty fucking useless.
 	equipslot = SLOT_NECK
 	attack_verb = null
 
+/obj/item/shadowcloak/magician/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/upgradewand))

```

Comment:
You should add a check to see if it is already upgraded before people buy 2 wands and try to double upgrade things, to a message saying they updated it with no actual effect.